client/src/main/java/io/pravega/client/connection/impl/AppendBatchSizeTrackerImpl.java

```diff
@@ -89,8 +89,7 @@ public int getAppendBlockSize() {
         }
         double nanosPerAppend = nanosBetweenAppends.getCurrentValue();
         double appendsInMaxBatchTime = Math.max(1.0, (MAX_BATCH_TIME_MILLIS * NANOS_PER_MILLI) / nanosPerAppend);
-        double appendsInTime = Math.max(1.0, BASE_TIME_NANOS / nanosPerAppend);
-        double appendsInBatch = MathHelpers.minMax(appendsOutstanding.getCurrentValue() * OUTSTANDING_FRACTION + appendsInTime, 1.0, appendsInMaxBatchTime);
```

Comment:
If 1 event is a problem, perhaps a batch size of 0 would be better in these cases. So maybe change this 1 to 0.

Reply:
With the previous change, we do not need to touch this line.

Reply:
Yes, but only because your test sends the exact same event size every time. If there were some slight variation we would want this to be 0, otherwise we would have the same problem you observed.

Reply:
As I shown, the change you propose seems to do not have a big impact on latency, but the max throughput for small events is significantly reduced. For this reason, I would rather keep the algorithm in its current form (similar to what was before), as it has been tested as a good baseline in a variety of situations. We can address further improvements on it on separate PRs, as the purpose of this PR is only to solve the write latency problem for low event rates and multiple segments.